homeassistant/components/xiaomi_miio/gateway.py

```diff
@@ -45,3 +54,45 @@ async def async_connect_gateway(self, host, token):
             self._gateway_info.hardware_version,
         )
         return True
+
+
+class XiaomiGatewayDevice(Entity):
+    """Representation of a base Xiaomi Gateway Device."""
+
+    def __init__(self, subdevice):
+        """Initialize the Xiaomi Gateway Device."""
+        self._subdevice = subdevice
+        self._unique_id = subdevice.sid
+        self._name = subdevice.sid
```

Comment:
The problem is that a lot of people have more of the same devices, in my case I have 3 identical temperature sensors (indoor outdoor etc.) So if you base the name on only the type you will get identical names which makes things really confusing....

Idealy you want to get the names configured in the MiHome app, but I have tried verry hard but cannot get those names from the gateway, I just cant figure out the API command to retrieve those (same holds for the zigbee ID that I cant get from the gateway which is frustrating me tremendously, ther really schould be a API command for it, but just cant find it).

That is the reason I used the sid because that is unique for every device.
The entities are membors of an device which displays the device model so I though it was not nessesary to include the model again in the name, but I can definelty do that if you want.

Reply:
Yeah, I can see the issue with loads of similar devices, but simply having the sid won't tell anything to the user, either.. 

Maybe we should add the device name to SubDevice-inherited classes for the cases we know the name already? Something like what zigbee2mqtt does, i.e., my temperature sensor is marked in homeassistant as `MiJia temperature & humidity sensor (WSDCGQ01LM)`

Reply:
I could just simply use the device_type class name for this. that is accisible through the subdevice.device_type variable, but even better, I will add a property in the backend called name and then we can define the name in the backend.

Reply:
Name property (or `model_name` in case we have the possibility to get the name from the gateway in the future, in which case the `name` could return that?) sounds good to me.

Reply:
just added it in this very small PR: https://github.com/rytilahti/python-miio/pull/724

Reply:
this is done, just waiting on the new python_miio version